src/trace/span-inferrer.ts

```diff
@@ -112,12 +112,12 @@ export class SpanInferrer {
       endpoint: path,
       resource_names: resourceName,
       request_id: context?.awsRequestId,
```

Comment:
Via this branch I published v20 of dd lambda js to sandbox and asserted correct upstream service mapping for apigw and sns. The other services service.name are properly being added as shown by the spec file above

Reply:
<img width="1317" alt="Screenshot 2023-08-23 at 5 09 50 PM" src="https://github.com/DataDog/datadog-lambda-js/assets/49878080/ad00cd84-dbac-4ae3-a89c-257fcaa0565c">

Reply:
the issue why remapping wasnt working was the service.name is supported by backend and in the past it when it wasn't present it must of fallen back to the lambda function service name

Reply:
see https://github.com/search?q=repo%3ADataDog%2Fweb-ui%20service.name&type=code
https://github.com/search?q=repo%3ADataDog%2Fdd-trace-js%20service.name&type=code

Reply:
SLS-4019